Sources/Starknet/Providers/StarknetProvider/StarknetProvider.swift

```diff
@@ -40,126 +43,131 @@ public class StarknetProvider: StarknetProviderProtocol {
         return StarknetRequest<U>(method: method, params: params, config: config, networkProvider: networkProvider)
     }
 
-    public func batchRequests<U: Decodable>(requests: [StarknetRequest<U>]) throws -> StarknetBatchRequest<U> {
+    public func batchRequests<U>(requests: [any StarknetRequestProtocol<U>]) throws -> any StarknetBatchRequestProtocol<U> where U: Decodable {
         guard !requests.isEmpty else {
             throw StarknetProviderError.emptyBatchRequestError
         }
 
-        let rpcPayloads = requests.enumerated().map { index, request in
-            JsonRpcPayload(method: request.method, params: request.params, id: index)
+        let rpcPayloads = try requests.enumerated().map { index, request -> JsonRpcPayload in
+            guard let starknetRequest = request as? StarknetRequest<U> else {
+                throw StarknetProviderError.invalidBatchRequestTypeError
+            }
```

Comment:
This has been added to ensure that we can access `method` and `params` which is present in `StarknetRequest`.
If we didn't check the type here, `StarknetRequestProtocol` would have to include these two fields. Wdyt?

Reply:
In that case it probably makes more sense to have `method` and `params` as part of `StarknetRequestProtocol` instead of checking the type here.

Reply:
Agree, those should definitely be defined at the protocol level, they are not an implementation detail.